app/src/main/java/com/bunbeauty/papakarlo/feature/cafe/screen/cafe_list/CafeListFragment.kt

```diff
@@ -8,62 +8,107 @@ import androidx.compose.foundation.layout.padding
+import com.bunbeauty.papakarlo.common.ui.toolbar.FoodDeliveryToolbarScreen
 import com.bunbeauty.papakarlo.databinding.FragmentCafeListBinding
 import com.bunbeauty.papakarlo.extensions.setContentWithTheme
 import com.bunbeauty.papakarlo.feature.cafe.model.CafeItem
 import com.bunbeauty.papakarlo.feature.cafe.ui.CafeItem
+import com.bunbeauty.papakarlo.feature.product_details.ProductDetailsFragmentDirections
+import com.bunbeauty.papakarlo.feature.top_cart.TopCartUi
 import com.bunbeauty.shared.domain.model.cafe.CafeStatus
 import com.google.android.material.transition.MaterialFadeThrough
+import org.koin.android.ext.android.inject
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class CafeListFragment : BaseFragment(R.layout.fragment_cafe_list) {
 
     override val viewBinding by viewBinding(FragmentCafeListBinding::bind)
     override val viewModel: CafeListViewModel by viewModel()
+    private val cafeListUiStateMapper: CafeListUiStateMapper by inject()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         enterTransition = MaterialFadeThrough()
     }
 
+    @OptIn(ExperimentalLifecycleComposeApi::class)
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         overrideBackPressedCallback()
         super.onViewCreated(view, savedInstanceState)
 
         viewModel.getCafeItemList()
         viewBinding.fragmentCafeListCvMain.setContentWithTheme {
-            val cafeItemList by viewModel.cafeItemList.collectAsState()
-            CafeListScreen(cafeItemList)
+            val cafeItemList by viewModel.cafeListState.collectAsStateWithLifecycle()
+            CafeListScreen(
+                cafeListState = cafeListUiStateMapper.map(cafeItemList),
+                onCafeClicked = viewModel::onCafeCardClicked,
+                onRefreshClicked = viewModel::getCafeItemList
+            )
+            LaunchedEffect(cafeItemList.eventList) {
+                handleEventList(cafeItemList.eventList)
+            }
         }
     }
 
     @Composable
-    private fun CafeListScreen(cafeItemListState: State<List<CafeItem>>) {
-        when (cafeItemListState) {
-            is State.Success -> CafeListSuccessScreen(cafeItemListState.data)
-            is State.Loading -> LoadingScreen()
-            is State.Error -> {
-                ErrorScreen(mainTextId = R.string.error_cafe_list_loading) {
-                    viewModel.getCafeItemList()
+    private fun CafeListScreen(
+        cafeListState: CafeListUi,
+        onCafeClicked: (CafeItem) -> Unit,
+        onRefreshClicked: () -> Unit,
+    ) {
+        FoodDeliveryToolbarScreen(
+            title = stringResource(R.string.title_cafe_list),
+            topActions = listOf(
+                FoodDeliveryCartAction(
+                    topCartUi = cafeListState.topCartUi,
+                ) {
+                    val backQueue = findNavController().backQueue
```

Comment:
We don't need this logic here. Just `findNavController().navigate(ProductDetailsFragmentDirections.globalConsumerCartFragment())`